core/dbt/parser/manifest.py

```diff
@@ -497,8 +532,46 @@ def load(self):
             # write out the fully parsed manifest
             self.write_manifest_for_partial_parse()
 
+        self.check_for_model_deprecations()
+
         return self.manifest
 
+    def check_for_model_deprecations(self):
+        for node in self.manifest.nodes.values():
+            if isinstance(node, ModelNode):
+                if (
+                    node.deprecation_date
+                    and node.deprecation_date < datetime.datetime.now().astimezone()
```

Comment:
@peterallenwebb want to double-check two things with you:
1. naive `deprecation_date`s
2. single stable time

### naive `deprecation_date`s
I believe this would be dependent on the system time wherever the python is being executed. So we'll probably want to double-check that any `deprecation_date` expressed without an offset like `2023-05-17` is understood to have its offset from the system time zone.

### single stable time
Were you able to try out something akin to [`run_started_at`](https://docs.getdbt.com/reference/dbt-jinja-functions/run_started_at)?

It might be useful for using a single stable value over the course of the entire execution. It might also be useful from a testing perspective as well.